Updates for two mc releases (#1160)
minio/docs: source/reference/minio-mc/mc-put.rst

==========
``mc put``
==========

.. default-domain:: minio

.. contents:: Table of Contents
   :local:
   :depth: 2

.. mc:: mc put

.. versionadded:: mc RELEASE.2024-02-24T01-33-20Z

Syntax
------

.. start-mc-put-desc

The :mc:`mc put` uploads an object from the local file system to a bucket on a target S3 deployment.

.. end-mc-put-desc

Unlike other commands that can upload files, such as :mc:`mc cp` or :mc:`mc mirror`, ``mc put`` is a simpler call.

> ``mc put`` only does the one-way function of uploading the file so that it avoids the potential performance costs of other commands.

Reviewer: Same suggestion as mc get

Author: done

.. mc-cmd:: --encrypt
   :optional:

   Specify the key to use for decrypting and encrypting the uploaded object.

   Requires that you also specify the key to use with the :mc-cmd:`~mc put --encrypt-key` flag.

   Alternatively, set the :envvar:`MC_ENCRYPT` environment variable.

.. mc-cmd:: --encrypt-key
   :optional:
   
   Specify the key to use for decrypting and encrypting the uploaded object.

   Requires that you also pass the :mc-cmd:`~mc put --encrypt` flag set to ``TRUE``.

   Alternatively, set the :envvar:`MC_ENCRYPT_KEY` environment variable.

.. mc-cmd:: --parallel, --P
   :optional:

   For multi-part uploads, specify the number of parts of the object to upload in parallel.

   If not defined, defaults to a value of ``4``.

.. mc-cmd:: --part-size, -s
   :optional:

   Specify the size to use for each part of a multi-part upload.

   If not defined, defaults to a value of ``16MiB``.

Global Flags
~~~~~~~~~~~~

.. include:: /includes/common-minio-mc.rst
   :start-after: start-minio-mc-globals
   :end-before: end-minio-mc-globals

Examples
--------

Upload a File and Specify the Object Name
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The following command uploads the file ``logo.png`` from the local file system to the ``business`` bucket on the ``minio`` deployment with the object name of ``company-logo.png``.

Reviewer 2: maybe
```suggestion
The following command uploads the file ``logo.png`` from the local file system to the ``business`` bucket on the ``minio`` deployment, renaming the object to ``company-logo.png``.
```

Author: Hesitant to introduce possible confusion that it is renaming the local copy, which it is not.

Author: But trying a different turn of phrase.

.. code-block:: shell
   :class: copyable

   mc put images/collateral/logo.png minio/business/company-logo.png

> Upload a File 8 Parts in Parallel with a Specified Part Size

Reviewer 2: I think this means it breaks the file into 8 parts for upload? Perhaps
```suggestion
Upload a File as 8 Parts in Parallel, using a Specified Part Size
```
Although I wonder what happens if the part size and number of parts doesn't neatly match the file size.

Author: No, the file is chunked into however many parts make it fit the specified part size.
This example uploads 8 of those parts in parallel. Will tweak.
